TypeCobol/Compiler/Types/ArrayType.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.Symbols;
+using static TypeCobol.Compiler.Symbols.Symbol;
+
+namespace TypeCobol.Compiler.Types
+{
+    /// <summary>
+    /// Class that represents an Array Type
+    /// </summary>
+    public class ArrayType : Type
```

Comment:
Do we need an ArrayType ?
I know we have this "type" in existing DataType class, but I think it was a mistake.

As OCCURS clause can be mixed with Group item, Picture item, Type item, pointer item, ... I'd rather see OCCURS as a property of a VariableSymbol.

Reply:
We need an ArrayType, for a Type System, to capture it as a real type. An ArrayType is determinated based on the MinOccurencesCount and the MaxOccurencesCount properties of a DataDefinition.

Reply:
After discussion, the ArrayType will always be used in combination with another type.